File: javascript/7_kyu/diagonals_sum.js
// Diagonals sum
// https://www.codewars.com/kata/5592fc599a7f40adac0000a8/train/javascript

function sum(matrix) {
    if (matrix.length == 0) {
        return 0;
    }

    var ans = [];
    var x = 0;
    var y = matrix[0].length - 1;
    
    for (var i = 0; i < matrix.length; i++) {
      ans.push(matrix[i][x]);
      ans.push(matrix[i][y]);
      x++;
      y--;
    }
    
    return ans.reduce(function (a, b) { return a + b });
  }

  console.log(sum([[1,2,3], [4,5,6], [7,8,9]]))

File: python/7_kyu/diagonals_sum.py
# Diagonals sum
# https://www.codewars.com/kata/5592fc599a7f40adac0000a8/train/python

def sum_diagonals(matrix):
    ans = []
    x = 0
    y = -1

    for i in matrix:
        ans.append(i[x])
        ans.append(i[y])
        x += 1
        y -= 1

    return sum(ans)

print(sum_diagonals([[1,2,3], [4,5,6], [7,8,9]]))